app/layout/account-bar.jsx

```diff
@@ -22,92 +23,83 @@ counterpart.registerTranslations('en', {
     settings: 'Settings',
     signOut: 'Sign Out',
     collections: 'Collections',
-    favorites: 'Favorites',
-  },
+    favorites: 'Favorites'
+  }
 });
 
-const AccountBar = React.createClass({
-  contextTypes: {
-    user: React.PropTypes.object,
-    router: routerShape,
-    geordi: React.PropTypes.object,
-  },
-
-  getInitialState() {
-    return {
-      unread: false,
+class AccountBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.navigateMenu = this.navigateMenu.bind(this);
+    this.handleSignOutClick = this.handleSignOutClick.bind(this);
+    this.lookUpUnread = this.lookUpUnread.bind(this);
+    this.state = {
+      unread: false
     };
-  },
+  }
 
   componentDidMount() {
     addEventListener('locationchange', this.lookUpUnread);
     this.lookUpUnread();
-  },
+  }
 
   componentWillReceiveProps(nextProps, nextContext) {
     this.logClick = !!nextContext &&
       !!nextContext.geordi &&
       !!nextContext.geordi.makeHandler &&
       nextContext.geordi.makeHandler('about-menu');
-  },
+  }
 
   componentWillUnmount() {
     removeEventListener('locationchange', this.lookUpUnread);
-  },
+  }
 
   lookUpUnread() {
     talkClient.type('conversations').get({
       user_id: this.context.user.id,
       unread: true,
-      page_size: 1,
+      page_size: 1
     }).then((conversations) => {
       this.setState({
-        unread: conversations.length > 0,
+        unread: conversations.length > 0
       });
     });
-  },
-
-  handleAccountMenuOpen() {
-    setTimeout(() => { // Wait for the modal's internal state change to happen.
-      if (this.refs.accountMenuButton.state.open) {
-        // React's `autoFocus` apparently doesn't work on <a> tags.
-        const firstFocusable = this.refs.accountMenu.querySelector(FOCUSABLES);
-        if (!!firstFocusable) {
-          firstFocusable.focus();
-        }
-      }
-    });
-  },
+  }
 
   navigateMenu(event) {
-    const focusables = this.refs.accountMenu.querySelectorAll(FOCUSABLES);
-    const focusIndex = Array.prototype.indexOf.call(focusables, document.activeElement);
+    const focusables = [ReactDOM.findDOMNode(this.accountMenuButton)];
+    if (this.accountMenu) {
+      for (const item of this.accountMenu.querySelectorAll(FOCUSABLES)) {
+        focusables.push(item);
+      }
+    }
+    const focusIndex = focusables.indexOf(document.activeElement);
 
     const newIndex = {
       [UP]: Math.max(0, focusIndex - 1),
-      [DOWN]: Math.min(focusables.length - 1, focusIndex + 1),
+      [DOWN]: Math.min(focusables.length - 1, focusIndex + 1)
     }[event.which];
```

Comment:
It's supposed to be supported from IE9+ but might be returning a different code for the up/down arrows.